content/articles/searching-and-sorting-techniques/index.md

```diff
@@ -0,0 +1,231 @@
+
+Sorting, for example, is required before using Binary Search, and it is also utilized in Database methods. Sorted arrays and lists make it easier to locate items fast. In this article, we will learn about various sorting and searching techniques and their implementation in data structures and algorithms. 
+### Table of Contents
+- [What is searching?](#what-is-searching)
+- [What is sorting?](#what-is-sorting)
+- [Techniques for searching and how they are implemented ](#techniques-for-searching-and-how-they-are-implemented )
+  1. [Linear Search](#1-linear-search)
+  2. [Binary Search](#2-binary-search)
+  3. [Interpolation Search](#3-interpolation-search)
+  4. [Jump Search](#2-binary-search)
+- [Sorting techniques and their implementation](#sorting-techniques-and-their-implementation) 
+   1. [Insertion Sort](#1-insertion-sort)
+   2. [Bubble Sort](#2-bubble-sort)
+   3. [Selection Sort](#3-selection-sort)
+   4. [Merge Sort](#4-merge-sort)
+   5. [Quick Sort](#5-quick-sort)
+   6. [Heap Sort](#6-heap-sort)
+   7. [Conclusion](#conclusion)
+### What is searching?
+Searching in data structure refers to the act of locating needed information from a group of things stored in the form of components in the computer memory.
+### What is sorting?
+Sorting refers to the operation or technique of ordering and rearranging groups of data in a specified order.
+### Techniques for searching and how they are implemented 
+These techniques are divided into categories depending on the kind of search operation they undertake, for example:
+#### 1. Linear Search
+The linear search method is a pretty simple search method. A sequential search is performed on all things one by one in this sort of search. Every item is verified, and if a match is discovered, that item is returned; if not, the search continues until the data collection is complete.
+##### Algorithm
+step 1:  Determine the size of the data set.
+
+step 2: Set the counter to zero.
+
+step 3: Examine the value in the counter position of the list.
+
+step 4: Examine to see if the value at that place corresponds to the value you're searching for.
+
+step 5: If it matches, the value has been discovered. Send a message to bring the search to a close.
+
+step 6: If not, go return to step 3 and increase the counter by one until there are no more things to search for.
+
+step 7: Send a message if all of the objects have been searched and no match has been found.
+#### 2. Binary Search
+To search a sorted array, divide the search interval in half regularly. Begin by creating an interval that spans the whole array. If the search key's value is less than the item in the interval's midpoint, the interval should be narrowed to the bottom half. Otherwise, limit it to the upper half of the page. Check the value until it is discovered or the interval is empty.

```

Comment:
```suggestion
To search a sorted array, divide the search interval in half regularly. Begin by creating an interval that spans the whole array. If the search key's value is less than the item in the interval's midpoint, the interval should be narrowed to the bottom half. Otherwise, limit it to the upper half of the page. Check the value until it is discovered or the interval is empty.

```